frontend/src/Elements/components/Consent/Consent.js

```diff
@@ -6,35 +6,29 @@ import _ from 'lodash';
 import Checkbox from '../Common/Checkbox';
 import PopoverView from '../Common/Views/Popover';
 import ProcessorsList from '../Processors/ProcessorsList';
+import Subject from '../../contexts/Subject';
 
 class Consent extends React.PureComponent {
   state = {
     processors: [],
     showProcessors: false
   };
 
+  consentGiven = false;
+
   async componentDidMount() {
-    if (!this.state.processors.lenght) {
-      try {
-        const processors = await this.props.cg.Subject.getProcessors();
-        this.setState({
-          processors: processors.map(p => {
-            p.enabled = true;
-            return p;
-          })
-        });
-      } catch (e) {
-        console.log('failure', e);
-      }
-    }
+    await this.props.subject.fetchProcessors();
+    const { processors, isGuest } = this.props.subject;
+    this.setState({ processors, isGuest });
 
     this.form = ReactDOM.findDOMNode(this).parentNode;
-    this.form.addEventListener('submit', this.waitForToken, false);
   }
 
-  componentWillUnmount() {
-    this.form.removeEventListener('submit', () => {});
-    this.props.cg.Events.clear('auth.setAccessToken');
+  componentWillUpdate(newProps) {
+    //@todo figure out a better way how to listen to subject's sign up event
```

Comment:
We have support for events on `js-sdk` actions, perhaps we can reuse that, or we need to add a subscriber for Elements SDK properly. We can do it in a separated task. Will be good to discuss the approach

Old way: `this.cg.Events.subscribe('auth.setAccessToken', () => {})`